src/types/composer.ts

```diff
@@ -1409,692 +1368,634 @@ export class TransactionComposer {
    * ```
    */
   addOfflineKeyRegistration(params: OfflineKeyRegistrationParams): TransactionComposer {
-    this.txns.push({ ...params, type: 'keyReg' })
+    this.validateGroupSize(1)
+
+    this.txns.push({ data: params, type: 'keyReg' })
 
     return this
   }
 
   /**
-   * Add the transactions within an `AtomicTransactionComposer` to the transaction group.
-   * @param atc The `AtomicTransactionComposer` to build transactions from and add to the group
-   * @returns The composer so you can chain method calls
+   * Get the number of transactions currently added to this composer.
+   * @returns The number of transactions currently added to this composer
+   */
+  count() {
+    return this.txns.length
+  }
+
+  /**
+   * Compose all of the transactions in a single transaction group and an transaction composer.
+   *
+   * You can then use the transactions standalone, or use the composer to execute or simulate the transactions.
+   *
+   * Once this method is called, no further transactions will be able to be added.
+   * You can safely call this method multiple times to get the same result.
+   * @returns The built transaction composer, the transactions and any corresponding method calls
    * @example
    * ```typescript
-   * const atc = new AtomicTransactionComposer()
-   *   .addPayment({ sender: 'SENDERADDRESS', receiver: 'RECEIVERADDRESS', amount: 1000n })
-   * composer.addAtc(atc)
+   * const { atc, transactions, methodCalls } = await composer.build()
    * ```
    */
-  addAtc(atc: algosdk.AtomicTransactionComposer): TransactionComposer {
-    this.txns.push({ atc, type: 'atc' })
-    return this
-  }
-
-  /** Build an ATC and return transactions ready to be incorporated into a broader set of transactions this composer is composing */
-  private buildAtc(atc: algosdk.AtomicTransactionComposer): TransactionWithSignerAndContext[] {
-    const group = atc.buildGroup()
-
-    const txnWithSigners = group.map((ts, idx) => {
-      // Remove underlying group ID from the transaction since it will be re-grouped when this TransactionComposer is built
-      ts.txn.group = undefined
-      // If this was a method call return the ABIMethod for later
-      if (atc['methodCalls'].get(idx)) {
-        return {
-          ...ts,
-          context: { abiMethod: atc['methodCalls'].get(idx) as algosdk.ABIMethod },
-        }
-      }
-      return {
-        ...ts,
-        context: {},
-      }
-    })
-
-    return txnWithSigners
-  }
+  public async build() {
+    if (!this.transactionsWithSigners) {
+      const suggestedParams = await this.getSuggestedParams()
+      const builtTransactions = await this.buildTransactionsSuggestedParamsProvided(suggestedParams)
```

Comment:
```suggestion
      const builtTransactions = await this._buildTransactions(suggestedParams)
```

If you're just trying to have a different name.